morpheus/_lib/src/stages/kafka_source.cpp

```diff
@@ -74,6 +79,24 @@
 #endif  // DOXYGEN_SHOULD_SKIP_THIS
 
 namespace morpheus {
+
+KafkaOAuthCallback::KafkaOAuthCallback(const std::function<std::map<std::string, std::string>()>& oauth_callback) :
+  m_oauth_callback(oauth_callback)
+{}
+
+void KafkaOAuthCallback::oauthbearer_token_refresh_cb(RdKafka::Handle* handle, const std::string& oauthbearer_config)
+{
+    auto response = m_oauth_callback();
```

Comment:
Lets add a try/catch block here to add more information about any exceptions that were generated.